lib/srv/sess.go

```diff
@@ -1360,6 +1383,9 @@ func (s *session) startExec(ctx context.Context, channel ssh.Channel, scx *Serve
 
 		s.emitSessionEndEvent()
 		s.Close()
+
+		s.io.Close()
+		close(s.doneCh)
```

Comment:
Was this always missing or is this new because of the assist changes?

Reply:
I was never needed as non interactive session was never recorded. This is also related to https://github.com/gravitational/teleport/issues/21267, which shows that it's not fully reliable.